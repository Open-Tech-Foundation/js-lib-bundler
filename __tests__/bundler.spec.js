/* eslint-disable @typescript-eslint/no-var-requires */
const { rmSync, readFileSync } = require('fs');
const { tmpdir } = require('os');
const { join } = require('path');
const { bundler } = require('../dist/bundler.cjs');
const jsonToFiles = require('./jsonToFiles.js');

const tempDir = join(tmpdir(), 'js-lib-bundler', 'lib');

beforeEach(() => {
  const cwdSpy = jest.spyOn(process, 'cwd');
  cwdSpy.mockReturnValue(tempDir);
  rmSync(tempDir, { recursive: true, force: true });
});

describe('Bundler', () => {
  test('Single JS file with an expression', () => {
    jsonToFiles(tempDir, {
      'a.js': `console.log("Hello World!")`,
    });
    bundler({ source: 'a.js', exports: './main.js' });
    const output = readFileSync(join(tempDir, 'main.js'), {
      encoding: 'utf-8',
    });
    expect(output).toMatchSnapshot();
  });

  test('Single JS file with a function', () => {
    jsonToFiles(tempDir, {
      'a.js': `function a() { return "Hello World!" }`,
    });
    bundler({ source: 'a.js', exports: './main.js' });
    const output = readFileSync(join(tempDir, 'main.js'), {
      encoding: 'utf-8',
    });
    expect(output).toMatchSnapshot();
  });

  test('Single JS file with export default function', () => {
    jsonToFiles(tempDir, {
      'a.js': `export default function a() { return "Hello World!" }`,
    });
    bundler({ source: 'a.js', exports: './main.js' });
    const output = readFileSync(join(tempDir, 'main.js'), {
      encoding: 'utf-8',
    });
    expect(output).toMatchSnapshot();
  });

  test('Single JS file with a function & seperate default exp', () => {
    jsonToFiles(tempDir, {
      'a.js': `function a() { 
        return "Hello World!" 
      }
      
      export default a
      `,
    });
    bundler({ source: 'a.js', exports: './main.js' });
    const output = readFileSync(join(tempDir, 'main.js'), {
      encoding: 'utf-8',
    });
    expect(output).toMatchSnapshot();
  });

  test('Single JS file with a function & named export', () => {
    jsonToFiles(tempDir, {
      'a.js': `function a() { 
        return "Hello World!" 
      }
      
      export {a}
      `,
    });
    bundler({ source: 'a.js', exports: './main.js' });
    const output = readFileSync(join(tempDir, 'main.js'), {
      encoding: 'utf-8',
    });
    expect(output).toMatchSnapshot();
  });
});
