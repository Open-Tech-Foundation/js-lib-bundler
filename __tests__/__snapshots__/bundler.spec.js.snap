// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Bundler Single JS file with a function & named export 1`] = `
"function a() {
    return \\"Hello World!\\";
}
export { a };
"
`;

exports[`Bundler Single JS file with a function & seperate default exp 1`] = `
"function a() {
    return \\"Hello World!\\";
}
export default a;
"
`;

exports[`Bundler Single JS file with a function 1`] = `
"\\"use strict\\";
function a() { return \\"Hello World!\\"; }
"
`;

exports[`Bundler Single JS file with an expression 1`] = `
"\\"use strict\\";
console.log(\\"Hello World!\\");
"
`;

exports[`Bundler Single JS file with export default function 1`] = `
"export default function a() { return \\"Hello World!\\"; }
"
`;
